### ISSUE TITLE
Creating memory and temporary databases

### ISSUE THREAD

**Issue by houghtonap** [pawelsalawa/sqlitestudio#4064]
### Details
1. It is not obvious how to create memory databases in the User Interface.
2. There doesn't appear to be a way to create a temporary database in the User Interface.

### Steps to reproduce
1. Click Database
2. Click Add a database
3. Database dialog box provides database type, file, schema name, permanence option.

For memory databases it is possible to place `:memory:` into the File text box to create one, but that is not obvious to a user who is not familiar with the underlying SQLite interface. A suggestion would be to make memory databases a selectable Database type in the drop down, but that may cause issues since you may need an entry for SQLite 3 Memory Database, SQLCipher Memory Database, WxSQLite3 Memory Database.

For temporary databases it is not possible to have an empty File text box. Temporary databases in the SQLite database API are created using an empty string. See [Temporary Databases](https://sqlite.org/inmemorydb.html) in the SQLite Manual. Again, a suggestion would be make temporary databases a selectable Database type in the drop down, but this presents the same issue with Memory Databases and needing a separate entry for Database type.

However, a better solution for the Database dialog box might be:
* Database Format drop down with choices: SQLite 3, SQLCipher, WxSQLite 3.
* Database Type drop down with choices: File, Memory, Temporary
* File Location text box that is only active when Database Type is File
* Schema Name (on the list) text box allowing you to enter a schema name.
* Options section as it currently is.
* Test Connection button as it currently is.

Oddities with Memory Databases. When you set the option to persist a Memory database no warning is issued, which is fine for me, probably not for a novice, but when you exit SQLiteStudio and relaunch it, you get an error on the persisted Memory database and basically are forced to delete and add again. This is probably because the current version didn't think about the `:memory:` trick in the File text box. Here are some further suggestions:

1. Issue a warning when a user creates a memory database and they check the box to persist it, which indicates the memory database will be lost upon exit unless it has been exported to SQL before exiting the application.
2. Upon launching, check to see if there are persisted memory databases, create empty ones with their given schema names and attach/connect to them.
3. Further it might also be nice to allow an initial SQL editor script to be run which then could re-populate the memory database or do other setup.
   * It would be nice if SQL editor scripts could be saved in your own database like the special SQLite system tables are. This would allow a specially named SQL editor script (maybe named on-connect as opposed to on-disconnect) to be automatically executed after the database was connected to and would also allow a special item to be placed in tree under the database (maybe Scripts), like Tables, Views, etc., that would open the editing window where you could run/edit the named script.
   * ~~It would be nice to have global SQL editor scripts, not associated with a specific database, saved in SQLite database which is automatically attached/connected at application launch.~~

### Operating system
Windows 7

### SQLiteStudio version
3.3.2

**Comment by pawelsalawa** [comment]
I allowed myself to strikethrough the last point and move it into separate ticket (as it's independent idea): #4065

**Comment by pawelsalawa** [comment]
As for the scripts per in-memory database I think it should not be placed next to tables, etc, because these are expanded and made available only after you open the database.

The initialization scripts (which I understand you need here) should be available rather from the properties dialog of the database, where you specify db type, file (or in-memory/temporary) and also you could click to edit init SQL scripts.

The scripts would be kept in SQLiteStudio's configuration, not in the in-memory database, because, well, in-memory database would lose it every time.
